dartagnan/src/main/java/com/dat3m/dartagnan/verification/RefinementTask.java

```diff
@@ -64,45 +101,39 @@ public BooleanFormula encodeBaselineWmmConsistency(SolverContext ctx) {
     public void initialiseEncoding(SolverContext ctx) {
         super.initialiseEncoding(ctx);
         baselineModel.initialise(this, ctx);
+		logger.info("{}: {}", ASSUME_LOCALLY_CONSISTENT_WMM, useLocallyConsistentBaselineWmm);
+		logger.info("{}: {}", ASSUME_NO_OOTA, useNoOOTABaselineWMM);
     }
 
     public static RefinementTask fromVerificationTaskWithDefaultBaselineWMM(VerificationTask task) {
-        return new RefinementTask(
-                task.getProgram(),
-                task.getMemoryModel(),
-                createDefaultWmm(),
-                task.getWitness(),
-                task.getTarget(),
-                task.getSettings()
-        );
+        return new RefinementTaskBuilder()
+                .withWitness(task.getWitness())
+                .withConfig(task.getConfig())
+                .build(task.getProgram(),task.getMemoryModel());
     }
 
-    private static Wmm createDefaultWmm() {
+    //TODO: This code is outdated and was replaced in the CAAT branch.
```

Comment:
Can we remove then the method?

Reply:
No, it needs to get updated with the code that sits in development.

Reply:
You fixed this only partially. In the development version we could have enabled all of the various baseline options simultanously, but in the new version you have to decide for a single one (it is not possible to have uniproc + oota).
A completely different alternative would be to allow a `.cat` file to be specified as baseline wmm, but then we need to provide some `.cat` files that can be used for refinement.
Either way, we can keep it as it is for now and try to extend this option later. We run refinement without any baseline model most of the time anyways.

Reply:
Now you can assign any of these 3 options. The general solution using `.cat` could be nice, but until we have some empirical evidence on how useful this is (even just with the 3 options we have) I would not go in that direction.